Proteomics/ProteolyticDigestion/PeptideWithSetModifications.cs

```diff
@@ -96,7 +99,41 @@ public double MonoisotopicMass
             {
                 if (!_monoisotopicMass.HasValue)
                 {
-                    double monoMass = WaterMonoisotopicMass;
+                    double monoMass = 0;
```

Comment:
can you explain what's going on here? if there are mods annotated in the sequence then you shouldn't need special code to calculate the mono mass right?

Reply:
So for traditional peptide generation we add the monoisotopic mass of water to the mass of all the amino acid residues because there is an -OH that caps the c-terminus and an -OH that caps the n-terminus.
Modifications that occur because of proteolytic cleavage can affect this. the homo-serine lactone formed by CNBr cleavage is a great example of this in that it would be incorrect to have the OH group on the C-terminus and that would make the mass incorrect. To remedy this. The protease modification takes in information as to whether or not the N-terminal H and C-terminal OH should be maintained for peptide cleavage. The changes here ensure that the mass calculations are handled properly.  
Its not per say about the mod itself its about the affect the mod has on the "finishing" of the peptide sequence. 
It would also not be entirely correct to have the calculated into the PTM since it is not actually the mass of the modification.